src/librustc_codegen_ssa/mir/place.rs

```diff
@@ -254,14 +254,15 @@ impl<'a, 'tcx, V: CodegenObject> PlaceRef<'tcx, V> {
                 };
                 let relative_max = niche_variants.end().as_u32() - niche_variants.start().as_u32();
                 let is_niche = {
-                    let relative_max = if relative_max == 0 {
+                    if relative_max == 0 {
```

Comment:
You can remove the extra block, it only existed for the large `let relative_max = ...;`.